Sources/Sentry/SentryThreadInspector.m

```diff
@@ -12,6 +18,36 @@
 
 @end
 
+typedef struct {
+    SentryCrashThread thread;
+    SentryCrashStackEntry stackEntries[MAX_STACKTRACE_LENGTH];
+    int stackLength;
+} ThreadInfo;
+
+// We need a C function to retrieve information from the stack trace in order to avoid
+// calling into async code while there is suspended threads.
+unsigned int
+getStackEntriesFromThread(SentryCrashThread thread, struct SentryCrashMachineContext *context,
+    SentryCrashStackEntry *buffer, unsigned int amount)
+{
+    sentrycrashmc_getContextForThread(thread, context, false);
+    SentryCrashStackCursor stackCursor;
+    sentrycrashsc_initWithMachineContext(&stackCursor, MAX_STACKTRACE_LENGTH, context);
+
+    unsigned int result = 0;
+    while (stackCursor.advanceCursor(&stackCursor)) {
+        if (result == amount)
+            break;
+        if (stackCursor.symbolicate(&stackCursor)) {
```

Comment:
`m`: Don't we also need a check for the marker frame here as we do in 
https://github.com/getsentry/sentry-cocoa/blob/17a7a28bf01bb7c7a095bbd0cded5e6bc1dd65aa/Sources/Sentry/SentryStacktraceBuilder.m#L38-L51

Reply:
We do this later when building the stacktrace [here](https://github.com/getsentry/sentry-cocoa/blob/17a7a28bf01bb7c7a095bbd0cded5e6bc1dd65aa/Sources/Sentry/SentryStacktraceBuilder.m#L62-L75). At this moment we don't have a way register if the frame is a stackStart.